src/+replab/Rational.m

```diff
@@ -0,0 +1,464 @@
+            if g ~= 1
+                R.N = R.N ./ g;
+                R.d = R.d / g;
+            end
+        end
+
+        function M = double(R)
+            M = R.N/R.d;
+        end
+
+        function R1 = diag(R)
+            R1 = replab.Rational(diag(R.N), R.d);
+        end
+
+        function R = mtimes(A, B)
+            if isa(A, 'sym') || isa(B, 'sym')
+                R = sym(A) * sym(B);
+            else
+                A = replab.Rational(A);
+                B = replab.Rational(B);
+                R = simplify(replab.Rational(A.N * B.N, A.d * B.d));
+            end
+        end
+
+        function R = times(A, B)
+            if isa(A, 'sym') || isa(B, 'sym')
+                R = sym(A) .* sym(B);
+            else
+                A = replab.Rational(A);
+                B = replab.Rational(B);
+                R = simplify(replab.Rational(A.N .* B.N, A.d * B.d));
+            end
+        end
+
+        function [root, res] = sqrt(R)
+        % Rational square root. root.^2 * res == R
+        %
+        % TODO: what does this do?
+            R1 = simplify(R);
+            droot = 1;
+            dres = 1;
+            d = R.d;
+            f = unique(factor(d));
+            for i = f
+                while gcd(d, i*i) > 1
+                    d = d / (i*i);
+                    droot = droot * i;
+                end
+                while gcd(d, i) > 1
+                    d = d / i;
+                    dres = dres * i;
+                end
+            end
+            N = R.N;
+            Nroot = zeros(size(N));
+            Nres = zeros(size(N));
+            Nroot(N ~= 0) = 1;
+            Nres(N > 0) = 1;
+            Nres(N < 0) = -1;
+            N = abs(N);
+            f = unique(factor(abs(replab.Rational.matlcm(N))));
+            for i = f
+                while gcd(replab.Rational.matlcm(N), i*i) == i*i && i > 1
+                    ind = (mod(N, i*i) == 0) & (N ~= 0);
+                    N(ind) = N(ind) / (i*i);
+                    Nroot(ind) = Nroot(ind) * i;
+                end
+                while gcd(replab.Rational.matlcm(N), i) == i && i > 1
+                    ind = (mod(N, i) == 0) & (N ~= 0);
+                    N(ind) = N(ind) / i;
+                    Nres(ind) = Nres(ind) * i;
+                end
+            end
+            root = replab.Rational(Nroot, droot);
+            res = replab.Rational(Nres, dres);
+        end
+
+        function R = power(R1, n)
+            R = replab.Rational(R1.N.^n, R1.d.^n);
+        end
+
+        function R = transpose(R)
+            R.N = R.N';
+        end
+
+        function R = ctranspose(R)
+            R.N = R.N';
+        end
+
+        function R = rref(R)
+            R.N = rref(R.N);
+            R.d = 1;
+        end
+
+        function R = permute(R, order)
+            R.N = permute(R.N, order);
+        end
+
+        function R = squeeze(R)
+            R.N = squeeze(R.N);
+        end
+
+        function c = char(R)
```

Comment:
On a generic double, `char(3.4)` returns `''`, so this looks more like the `num2str` function. Should we add it?